packages/features/bookings/components/FiltersContainer.tsx

```diff
@@ -1,51 +1,196 @@
+        )}
+      </FilterCheckboxFieldsContainer>
+    </AnimatedPopover>
+  );
 };
 
-const filters: Filter[] = [
-  {
-    name: "teams",
-    component: <TeamsMemberFilter />,
-    controllingQueryParams: ["teamId"],
-    showByDefault: true,
-  },
-  {
-    name: "people",
-    component: <PeopleFilter />,
-    controllingQueryParams: ["usersId"],
-    showByDefault: true,
-  },
-  {
-    name: "eventType",
-    component: <EventTypeFilter />,
-    controllingQueryParams: ["eventTypeId"],
-    showByDefault: true,
-  },
-];
+const LocationFilter = () => {
+  const { t } = useLocale();
+  const locations = trpc.viewer.locationOptions.useQuery();
+
+  const { data: query, pushItemToKey, removeItemByKeyAndValue, removeByKey } = useFilterQuery();
+
+  const getTextForPopover = () => {
+    const locationValues = query.locationValues;
+    if (locationValues) {
+      return `${t("location")}:  ${t("number_selected", { count: locationValues.length })}`;
+    }
+    return `${t("location")}: ${t("all")}`;
+  };
+
+  return (
+    <AnimatedPopover text={getTextForPopover()}>
+      <FilterCheckboxFieldsContainer>
+        {locations?.data?.map((location) => {
+          return (
+            <Fragment key={location.label}>
+              <div className="text-subtle py-2 px-4 text-xs font-medium uppercase leading-none">
+                {location.label}
+              </div>
+              {location.options?.map((option) => {
+                return (
+                  <FilterCheckboxField
+                    key={option.value}
+                    id={option.value}
+                    label={option.label}
+                    checked={!!query.locationValues?.includes(option.value)}
+                    onChange={(e) => {
+                      if (e.target.checked) {
+                        pushItemToKey("locationValues", option.value);
+                      } else if (!e.target.checked) {
+                        removeItemByKeyAndValue("locationValues", option.value);
+                      }
+                    }}
+                    icon={
+                      <img
+                        src={option.icon}
+                        alt="cover"
+                        // invert all the icons except app icons
+                        className={cx(
+                          "h-3.5 w-3.5",
+                          option.icon && !option.icon.startsWith("/app-store") && "dark:invert"
+                        )}
+                      />
+                    }
+                  />
+                );
+              })}
+            </Fragment>
+          );
+        })}
+      </FilterCheckboxFieldsContainer>
+    </AnimatedPopover>
+  );
+};
 
 export function FiltersContainer() {
+  const { t } = useLocale();
+
+  const [addFilterOptions, toggleOption, isFilterActive] = useBookingMultiFilterStore((state) => [
+    state.addFilterOptions,
+    state.toggleOption,
+    state.isFilterActive,
+  ]);
```

Comment:
Lets make sure these are shallow https://github.com/pmndrs/zustand#selecting-multiple-state-slices

or pick them one by one so we dont get any weird re-rendering issues

Reply:
Okay Thanks 👍